trunk/source/Library-Proofs/src/de/uni_freiburg/informatik/ultimate/lib/proofs/floydhoare/FloydHoareUtils.java

```diff
@@ -108,11 +109,18 @@ public static void createInvariantResults(final String pluginName, final IIcfg<I
 		final var checks = getCheckedSpecifications(icfg, annotation);
 
 		// find all locations that have outgoing edges which are annotated with LoopEntry, i.e., all loop candidates
-		final Set<IcfgLocation> locsForLoopLocations = new HashSet<>();
-		locsForLoopLocations.addAll(IcfgUtils.getPotentialCycleProgramPoints(icfg));
-		locsForLoopLocations.addAll(icfg.getLoopLocations());
-
-		for (final IcfgLocation locNode : locsForLoopLocations) {
+		final Set<IcfgLocation> locsForInvariants = new LinkedHashSet<>();
```

Comment:
Using a LinkedHashSet here to keep the order of 

1. Loop invariant results first
2. All other invariant results + their potential use inside a project second